test/unit/mongo_logger.test.ts

```diff
@@ -0,0 +1,347 @@
+import { expect } from 'chai';
+import { Writable } from 'stream';
+
+import {
+  MongoLogger,
+  MongoLoggerMongoClientOptions,
+  MongoLoggerOptions,
+  SeverityLevel
+} from '../../src/mongo_logger';
+
+class BufferingStream extends Writable {
+  buffer: string[] = [];
+
+  constructor(options = {}) {
+    super({ ...options, objectMode: true });
+  }
+
+  override _write(chunk, encoding, callback) {
+    this.buffer.push(chunk);
+    callback();
+  }
+}
+
+describe('meta tests for BufferingStream', function () {
+  it('the buffer is empty on construction', function () {
+    const stream = new BufferingStream();
+    expect(stream.buffer).to.have.lengthOf(0);
+  });
+  it('pushes messages to the buffer when written to', function () {
+    const stream = new BufferingStream();
+    stream.write('message');
+    expect(stream.buffer).to.deep.equal(['message']);
+  });
+});
+
+describe('class MongoLogger', function () {
+  describe('#constructor()', function () {
+    it('assigns each property from the options object onto the logging class', function () {
+      const componentSeverities: MongoLoggerOptions['componentSeverities'] = {
+        command: 'alert'
+      } as any;
+      const stream = new Writable();
+      const logger = new MongoLogger({
+        componentSeverities,
+        maxDocumentLength: 10,
+        logDestination: stream
+      });
+
+      expect(logger).to.have.property('componentSeverities', componentSeverities);
+      expect(logger).to.have.property('maxDocumentLength', 10);
+      expect(logger).to.have.property('logDestination', stream);
+    });
+  });
+
+  describe('static #resolveOptions()', function () {
+    describe('componentSeverities', function () {
```

Comment:
I did not table test the component severities - there was no straightforward table-test that I could think of.  I'm happy to consider alternate approaches though.

Reply:
it's fine as is, but I added a couple of notes on missing cases